src/vue/pages/SaleDetails.vue

```diff
@@ -0,0 +1,159 @@
+<template>
+  <div class="sale-details">
+    <top-bar v-if="sale.id">
+      <template slot="main">
+        <router-link
+          v-ripple
+          :to="vueRoutes.saleDetails"
+        >
+          <span>
+            {{ 'sale-details.campaign-title' | globalize }}
+          </span>
+        </router-link>
+      </template>
+
+      <template slot="extra">
+        <button
+          v-ripple
+          class="app__button-raised sale-details__invest-btn"
+          @click="isInvestDrawerShown = true"
+        >
+          {{ 'sale-details.invest' | globalize }}
+        </button>
+      </template>
+    </top-bar>
+
+    <drawer :is-shown.sync="isInvestDrawerShown">
+      <template slot="heading">
+        {{ 'sale-details.invest' | globalize }}
+      </template>
+
+      <invest-form
+        :sale="sale"
+        @submitted="refreshSale"
+        @canceled="refreshSale"
+      />
+    </drawer>
+
+    <template v-if="isLoaded">
+      <template v-if="sale.id">
+        <div class="sale-details__title">
+          <h2 class="sale-details__name">
+            {{ `${sale.name} (${sale.baseAsset})` }}
+          </h2>
+
+          <p class="sale-details__short-desc">
+            {{ sale.shortDescription }}
+          </p>
+        </div>
+
+        <sale-campaign
```

Comment:
it's `sale-campaign-viewer`, not the `sale-campaign` itself. The component name should tell us what exactly the component does

Reply:
shouldn't it be the separate route, according to design? you'll have a lot of `if`s when other details come

Reply:
Updated, renamed, added separate route